includes/admin/meta-boxes/class-wcs-meta-box-schedule.php

```diff
@@ -31,20 +31,25 @@ public static function output( $post ) {
 
 	/**
 	 * Save meta box data
+	 *
+	 * @see woocommerce_process_shop_order_meta
+	 *
+	 * @param int $post_or_order_id
+	 * @param WP_Post|WC_Order $post_or_order_object
 	 */
-	public static function save( $post_id, $post ) {
+	public static function save( $post_or_order_id, $post_or_order_object ) {
 
-		if ( 'shop_subscription' === $post->post_type && ! empty( $_POST['woocommerce_meta_nonce'] ) && wp_verify_nonce( wc_clean( wp_unslash( $_POST['woocommerce_meta_nonce'] ) ), 'woocommerce_save_data' ) ) {
+		if ( wcs_is_subscription( $post_or_order_id ) && ! empty( $_POST['woocommerce_meta_nonce'] ) && wp_verify_nonce( wc_clean( wp_unslash( $_POST['woocommerce_meta_nonce'] ) ), 'woocommerce_save_data' ) ) {
```

Comment:
I prefer to use early returns, e.g:

```php
if ( ! wcs_is_subscription( $post_or_order_id ) ) {
  return false;
}
```

Need to check if the returned value is of important to the `woocommerce_process_shop_order_meta` hook.

Reply:
Updated in 9debb2d